src/UBAFeeCalculator/UBAFeeSpokeCalculator.ts

```diff
@@ -60,17 +62,30 @@ export default class UBAFeeSpokeCalculator {
       return this.cachedRunningBalance[key];
     }
 
+    // Attempt to resolve the trigger hurdle to include in the running
+    // balance calculation
+    const triggerHurdle = this.getBalanceTriggerThreshold();
+
     // If the last validated running balance is undefined, we need to compute the running balance from scratch
     // This is the case when the UBA Fee Calculator is first initialized or run on a range
     // that we haven't computed the running balance for yet
     const historicalResult = this.recentRequestFlow.slice(startIdx, endIdx).reduce((acc, flow) => {
+      let resultantValue = acc;
       if (isUbaInflow(flow)) {
-        return acc.add(toBN(flow.amount));
+        resultantValue = acc.add(toBN(flow.amount));
       } else if (isUbaOutflow(flow)) {
-        return acc.sub(toBN(flow.amount));
-      } else {
-        return acc;
+        resultantValue = acc.sub(toBN(flow.amount));
+      }
+
+      // If the trigger hurdle is surpassed, we need to return the trigger hurdle value
+      // as the running balance. This is because the trigger hurdle is the maximum value we'd like to
+      // organically grow the running balance to. If the running balance exceeds the trigger hurdle,
+      // we need to return the trigger hurdle as the running balance because at this point the dataworker
+      // will be triggered to rebalance the running balance.
+      if (triggerHurdle !== undefined && resultantValue.gt(triggerHurdle[0])) {
```

Comment:
nit: unnamed tuples are a little hard to interpret -- have you considered using a named tuple or an object to represent these two values so it's clearer what's going on? 0 and 1 are not very self-documenting.

Reply:
Good call! https://github.com/across-protocol/sdk-v2/pull/196/commits/a6042c9971bc9aae70498c97ba80a6efe420518c